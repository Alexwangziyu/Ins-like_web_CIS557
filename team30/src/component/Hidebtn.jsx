/* eslint-disable react/prop-types */
/* eslint-disable react/prop-types */
import React, { useState, useEffect } from 'react';
import { Button } from 'antd';
import './SearchBar.css';
import { EyeInvisibleOutlined, EyeOutlined } from '@ant-design/icons';
import { addhide, delhide } from '../api/mock_api';
// import FavoriteBorderIcon from '@mui/icons-material/FavoriteBorder';

function hidebtn({
  myid, postid, setfunc, change,
}) {
  const [havehided, sethavehided] = useState(true);
  useEffect(() => {
    sethavehided(change.has(postid));
  });
  if (!havehided) {
    return (
      <Button
        type="link"
        className="likebtntest"
        data-testid="lbtn"
        icon={<EyeInvisibleOutlined style={{ fontSize: '100px' }} />}
        onClick={() => {
          addhide(postid, myid);
          const newset = new Set(change);
          newset.add(postid);
          setfunc(newset);
          // let a = true;
          // if (havehided) {
          //   a = false;
          // }
          // sethavehided(a);
        }}
      />
    );
  }
  return (
    <Button
      type="link"
      className="unlikebtntest"
      data-testid="unlbtn"
      icon={<EyeOutlined style={{ fontSize: '100px' }} />}
      onClick={() => {
        delhide(postid, myid);
        const newset = new Set(change);
        newset.delete(postid);
        setfunc(newset);
        // let a = true;
        // if (havehided) {
        //   a = false;
        // }
        // sethavehided(a);
      }}
    />
  );
}
export default hidebtn;
