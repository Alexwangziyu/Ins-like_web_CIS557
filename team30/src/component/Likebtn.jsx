/* eslint-disable react/prop-types */
import React, { useState, useEffect } from 'react';
import { Button } from 'antd';
import './SearchBar.css';
import { HeartOutlined, HeartFilled } from '@ant-design/icons';
import { haveliked, addlike, dellike } from '../api/mock_api';
// import FavoriteBorderIcon from '@mui/icons-material/FavoriteBorder';

function likebtn({ myid, postid }) {
  const [hasliked, setHasliked] = useState(false);
  useEffect(() => {
    haveliked(postid, myid).then((res) => setHasliked(res));
  }, [postid]);
  if (!hasliked) {
    return (
      <Button
        type="link"
        className="likebtntest"
        data-testid="lbtn"
        icon={<HeartOutlined style={{ fontSize: '100px' }} />}
        onClick={() => {
          setHasliked(!hasliked);
          addlike(postid, myid);
        }}
        danger
      />

    );
  }
  return (
    <Button
      type="link"
      className="unlikebtntest"
      data-testid="unlbtn"
      icon={<HeartFilled style={{ fontSize: '100px' }} />}
      onClick={() => {
        setHasliked(!hasliked);
        dellike(postid, myid);
      }}
      danger
    />
  );
}
export default likebtn;
