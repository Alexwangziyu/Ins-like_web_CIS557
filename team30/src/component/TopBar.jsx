/* eslint-disable jsx-a11y/alt-text */
/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/anchor-is-valid */
import React, { useState, useEffect } from 'react';
// Bootstrap CSS
import 'bootstrap/dist/css/bootstrap.min.css';
// Bootstrap Bundle JS
import 'bootstrap/dist/js/bootstrap.bundle.min';
import { useNavigate } from 'react-router-dom';
import { getUserID, getUser } from '../api/mock_api';
import logo from './logo.png';

function TopBar() {
  const navigate = useNavigate();
  const handleClick = (path) => {
    navigate(path);
  };
  const handleLogOut = () => {
    localStorage.removeItem('currentUser');
    sessionStorage.removeItem('id');
    sessionStorage.removeItem('app-token');
    navigate('/login');
  };
  const [udata, setudata] = useState({});
  const CurrentUserID = getUserID();
  useEffect(() => {
    // const CurrentUserID = getUserID();
    getUser(CurrentUserID).then((res) => setudata(res));
  }, []);

  // const Token = sessionStorage.getItem('app-token');

  return (
    <nav className="navbar navbar-expand-lg navbar-dark bg-primary bg-gradient shadow">
      <div className="container-fluid">
        <a className="navbar-brand" href="#">
          <img src={logo} width="115" height="40" alt="" />
        </a>
        <button className="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
          <span className="navbar-toggler-icon" />
        </button>
        <div className="collapse navbar-collapse" id="navbarSupportedContent">
          <ul className="navbar-nav me-auto mb-2 mb-lg-0">
            <li className="nav-item">
              <a className="nav-link active" aria-current="page" onClick={() => { handleClick(`/user/${CurrentUserID}`); }}>Activity</a>
            </li>
            <li className="nav-item">
              <a id="search" className="nav-link active" aria-current="page" onClick={() => { handleClick('/follow'); }}>Search</a>
            </li>
            <li className="nav-item dropdown">
              <a className="nav-link dropdown-toggle" href="#" id="navbarDropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
                Todo
              </a>
              <ul className="dropdown-menu" aria-labelledby="navbarDropdown">
                <li><a className="dropdown-item" onClick={() => { handleClick(`/profile/${CurrentUserID}`); }}>My Profile</a></li>
                <li><a className="dropdown-item" onClick={() => { handleClick('/post'); }}>New Post</a></li>
                <li><hr className="dropdown-divider" /></li>
                <li><a className="dropdown-item" onClick={() => { handleLogOut(); }}>Log Out</a></li>
              </ul>
            </li>
          </ul>
          <span className="navbar-text">
            <div className="navbar-brand marginleft=100px">{udata.name}</div>
          </span>
          <span className="navbar-brand">
            <a className="navbar-brand" href={`http://localhost:3000/profile/${CurrentUserID}`}>
              <img
                className="mb-0 rounded-circle border border-light border-2 display: inline"
                alt="showcase"
                style={{
                  width: '2.7rem', height: '2.7rem', objectFit: 'cover', borderRadius: '50%',
                }}
                src={udata.uavatar}
              />
            </a>
          </span>
        </div>
      </div>
    </nav>
  );
}

export default TopBar;
