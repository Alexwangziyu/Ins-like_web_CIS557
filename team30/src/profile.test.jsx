/* eslint-disable no-undef */
import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { act } from 'react-dom/test-utils';
import classNames from 'classnames';
import Profile from './Pages/Profile';
import TopBar from './component/TopBar';
import mockBackendAxios from './mock-backend-data';
import Likebtn from './component/Likebtn';
import Hidebtn from './component/Hidebtn';

mockBackendAxios();
describe('test Profile', () => {
  test('render <Profile /> normal case ', async () => {
    const response = await act(async () => renderer.create(
      <MemoryRouter>
        <Profile />
      </MemoryRouter>,
    ));
    expect(response.toJSON().length).toBe(5);
  });

  test('test click follow & unfollow btn', async () => {
    const response = await act(async () => renderer.create(
      <MemoryRouter>
        <Profile />
      </MemoryRouter>,
    ));
    console.log(response.toJSON().length);
    expect(response.toJSON().length).toBe(5);
  });

  test('profile matches snapshot', async () => {
    const component = await act(async () => renderer.create(
      <MemoryRouter>
        <Profile />
      </MemoryRouter>,
    ));
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('profile matches snapshot', () => {
    const component = renderer.create(
      <MemoryRouter>
        <TopBar />
      </MemoryRouter>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('test click topbar', async () => {
    const { container } = render(
      // eslint-disable-next-line react/jsx-filename-extension
      <MemoryRouter>
        <TopBar />
      </MemoryRouter>,
    );
    fireEvent.click(screen.getByText(/Activity/i));
    fireEvent.click(screen.getByText(/Search/i));
    fireEvent.click(screen.getByText(/Todo/i));
    expect(container.getElementsByClassName('dropdown-item')).toHaveLength(3);
    fireEvent.click(screen.getByText(/My Profile/i));
    fireEvent.click(screen.getByText(/New Post/i));
    fireEvent.click(screen.getByText(/Log Out/i));
  });

  test('profile matches snapshot', () => {
    const component = renderer.create(
      <MemoryRouter>
        <mock_api />
      </MemoryRouter>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('test Profile', () => {
  // eslint-disable-next-line no-undef
  test('render <Profile /> normal case ', async () => {
    // eslint-disable-next-line react/jsx-filename-extension
    const { container } = render(
      // eslint-disable-next-line react/jsx-filename-extension
      <MemoryRouter>
        <Profile />
      </MemoryRouter>,
    );
    const containerCls = classNames('container');
    const btnCls = classNames('btn btn-warning btn-sm active');
    // eslint-disable-next-line no-undef
    expect(container.getElementsByClassName(containerCls)).toHaveLength(5);
    // eslint-disable-next-line no-undef
    expect(container.getElementsByClassName(btnCls)).toHaveLength(1);
  });

  test('test click follow & unfollow btn', async () => {
    const { container } = render(
      // eslint-disable-next-line react/jsx-filename-extension
      <MemoryRouter>
        <Profile />
      </MemoryRouter>,
    );

    // init state
    expect(container.getElementsByClassName('followbtn')).toHaveLength(1);
    expect(container.getElementsByClassName('unfollowbtn')).toHaveLength(0);

    // click follow
    fireEvent.click(screen.getByText(/FOLLOW!/i));
    expect(container.getElementsByClassName('followbtn')).toHaveLength(0);
    expect(container.getElementsByClassName('unfollowbtn')).toHaveLength(1);

    // click unfollow
    fireEvent.click(screen.getByText(/UNFOLLOW!/i));
    expect(container.getElementsByClassName('followbtn')).toHaveLength(1);
    expect(container.getElementsByClassName('unfollowbtn')).toHaveLength(0);
  });

  test('profile matches snapshot', () => {
    const component = renderer.create(
      <MemoryRouter>
        <Profile />
      </MemoryRouter>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('likebtn matches snapshot', () => {
    const component = renderer.create(
      <MemoryRouter>
        <Likebtn />
      </MemoryRouter>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('hidebtn matches snapshot', () => {
    const component = renderer.create(
      <MemoryRouter>
        <Hidebtn />
      </MemoryRouter>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
