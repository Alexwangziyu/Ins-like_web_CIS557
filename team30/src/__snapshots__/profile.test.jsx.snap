// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Profile hidebtn matches snapshot 1`] = `
<button
  className="ant-btn ant-btn-link ant-btn-icon-only unlikebtntest"
  data-testid="unlbtn"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  <span
    aria-label="eye"
    className="anticon anticon-eye"
    role="img"
    style={
      Object {
        "fontSize": "100px",
      }
    }
  >
    <svg
      aria-hidden="true"
      data-icon="eye"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <path
        d="M942.2 486.2C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 000 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM512 766c-161.3 0-279.4-81.8-362.7-254C232.6 339.8 350.7 258 512 258c161.3 0 279.4 81.8 362.7 254C791.5 684.2 673.4 766 512 766zm-4-430c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm0 288c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
      />
    </svg>
  </span>
</button>
`;

exports[`test Profile likebtn matches snapshot 1`] = `
<button
  className="ant-btn ant-btn-link ant-btn-icon-only ant-btn-dangerous likebtntest"
  data-testid="lbtn"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  <span
    aria-label="heart"
    className="anticon anticon-heart"
    role="img"
    style={
      Object {
        "fontSize": "100px",
      }
    }
  >
    <svg
      aria-hidden="true"
      data-icon="heart"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <path
        d="M923 283.6a260.04 260.04 0 00-56.9-82.8 264.4 264.4 0 00-84-55.5A265.34 265.34 0 00679.7 125c-49.3 0-97.4 13.5-139.2 39-10 6.1-19.5 12.8-28.5 20.1-9-7.3-18.5-14-28.5-20.1-41.8-25.5-89.9-39-139.2-39-35.5 0-69.9 6.8-102.4 20.3-31.4 13-59.7 31.7-84 55.5a258.44 258.44 0 00-56.9 82.8c-13.9 32.3-21 66.6-21 101.9 0 33.3 6.8 68 20.3 103.3 11.3 29.5 27.5 60.1 48.2 91 32.8 48.9 77.9 99.9 133.9 151.6 92.8 85.7 184.7 144.9 188.6 147.3l23.7 15.2c10.5 6.7 24 6.7 34.5 0l23.7-15.2c3.9-2.5 95.7-61.6 188.6-147.3 56-51.7 101.1-102.7 133.9-151.6 20.7-30.9 37-61.5 48.2-91 13.5-35.3 20.3-70 20.3-103.3.1-35.3-7-69.6-20.9-101.9zM512 814.8S156 586.7 156 385.5C156 283.6 240.3 201 344.3 201c73.1 0 136.5 40.8 167.7 100.4C543.2 241.8 606.6 201 679.7 201c104 0 188.3 82.6 188.3 184.5 0 201.2-356 429.3-356 429.3z"
      />
    </svg>
  </span>
</button>
`;

exports[`test Profile profile matches snapshot 1`] = `
Array [
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-primary bg-gradient shadow"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand"
        href="#"
      >
        <img
          alt=""
          height="40"
          src="logo.png"
          width="115"
        />
      </a>
      <button
        aria-controls="navbarSupportedContent"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-bs-target="#navbarSupportedContent"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        id="navbarSupportedContent"
      >
        <ul
          className="navbar-nav me-auto mb-2 mb-lg-0"
        >
          <li
            className="nav-item"
          >
            <a
              aria-current="page"
              className="nav-link active"
              onClick={[Function]}
            >
              Activity
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              aria-current="page"
              className="nav-link active"
              id="search"
              onClick={[Function]}
            >
              Search
            </a>
          </li>
          <li
            className="nav-item dropdown"
          >
            <a
              aria-expanded="false"
              className="nav-link dropdown-toggle"
              data-bs-toggle="dropdown"
              href="#"
              id="navbarDropdown"
              role="button"
            >
              Todo
            </a>
            <ul
              aria-labelledby="navbarDropdown"
              className="dropdown-menu"
            >
              <li>
                <a
                  className="dropdown-item"
                  onClick={[Function]}
                >
                  My Profile
                </a>
              </li>
              <li>
                <a
                  className="dropdown-item"
                  onClick={[Function]}
                >
                  New Post
                </a>
              </li>
              <li>
                <hr
                  className="dropdown-divider"
                />
              </li>
              <li>
                <a
                  className="dropdown-item"
                  onClick={[Function]}
                >
                  Log Out
                </a>
              </li>
            </ul>
          </li>
        </ul>
        <span
          className="navbar-text"
        >
          <div
            className="navbar-brand marginleft=100px"
          >
            Error
          </div>
        </span>
        <span
          className="navbar-brand"
        >
          <a
            className="navbar-brand"
            href="http://localhost:3000/profile/null"
          >
            <img
              alt="showcase"
              className="mb-0 rounded-circle border border-light border-2 display: inline"
              style={
                Object {
                  "borderRadius": "50%",
                  "height": "2.7rem",
                  "objectFit": "cover",
                  "width": "2.7rem",
                }
              }
            />
          </a>
        </span>
      </div>
    </div>
  </nav>,
  <section
    className="p-5 bg-dark text-light"
  >
    <div
      className="container"
    >
      <div
        className="d-flex"
      >
        <img
          alt="showcase"
          loading="lazy"
          src="https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/971.jpg"
          style={
            Object {
              "borderRadius": "50%",
              "height": "16rem",
              "objectFit": "cover",
              "width": "16rem",
            }
          }
        />
        <div
          className="mx-4"
        >
          <h1>
            <span
              className="text-warning"
            >
              username1
            </span>
          </h1>
          <div>
            <h4
              className="text-warning"
            >
              follower:
            </h4>
            <h4
              className="text-warning"
            >
              62
            </h4>
          </div>
          <button
            aria-pressed="true"
            autoComplete="off"
            className="btn btn-warning btn-sm active unfollowbtn"
            data-bs-toggle="button"
            id="unfollowbtn"
            onClick={[Function]}
            type="button"
          >
            UNFOLLOW!
          </button>
          <p
            className="my-4"
          >
            Consequatur quo qui inventore. Nesciunt odio nisi nesciunt aperiam aut aut iusto deserunt voluptas. Occaecati voluptatibus saepedolores veritatis aut eum. Neque qui facilis earum eos velit numquam commodi sint ut.
          </p>
        </div>
      </div>
    </div>
    <div
      className="container bg-dark"
    >
      <div />
    </div>
  </section>,
  <div
    className="container"
  >
    <h1
      className="post1"
    >
      Posts:
    </h1>
  </div>,
  <section
    className="p-0"
  >
    <div
      className="container"
    >
      <ul
        className="MuiImageList-root MuiImageList-standard css-rvv586-MuiImageList-root"
        style={
          Object {
            "gap": 10,
            "gridTemplateColumns": "repeat(2, 1fr)",
          }
        }
      >
        <li
          className="MuiImageListItem-root MuiImageListItem-standard css-ta95xa-MuiImageListItem-root"
          style={
            Object {
              "gridColumnEnd": "span 3",
              "gridRowEnd": "span 1",
              "height": "auto",
              "marginBottom": undefined,
            }
          }
        />
        <li
          className="MuiImageListItem-root MuiImageListItem-standard css-ta95xa-MuiImageListItem-root"
          style={
            Object {
              "gridColumnEnd": "span 1",
              "gridRowEnd": "span 1",
              "height": "auto",
              "marginBottom": undefined,
            }
          }
        >
          <a
            href="http://localhost:3000/post/undefined"
          >
            <img
              alt="showcase"
              loading="lazy"
              src="https://res.cloudinary.com/dldiferrn/image/upload/v1667858043/darwin/cy6mqfl2zw6tvghltnwb.png?w=350&fit=crop&auto=format"
              style={
                Object {
                  "height": "26rem",
                  "objectFit": "cover",
                  "width": "26rem",
                }
              }
            />
          </a>
          <div
            className="MuiImageListItemBar-root MuiImageListItemBar-positionTop css-tdboxv-MuiImageListItemBar-root"
          >
            <div
              className="MuiImageListItemBar-titleWrap MuiImageListItemBar-titleWrapTop MuiImageListItemBar-titleWrapActionPosLeft css-3pkysr-MuiImageListItemBar-titleWrap"
            >
              <div
                className="MuiImageListItemBar-title css-dasnyc-MuiImageListItemBar-title"
              >
                   Should be a butter fly!
              </div>
            </div>
            <div
              className="MuiImageListItemBar-actionIcon MuiImageListItemBar-actionIconActionPosLeft css-1wnvjk9-MuiImageListItemBar-actionIcon"
            >
              <div />
            </div>
          </div>
        </li>
        <li
          className="MuiImageListItem-root MuiImageListItem-standard css-ta95xa-MuiImageListItem-root"
          style={
            Object {
              "gridColumnEnd": "span 1",
              "gridRowEnd": "span 1",
              "height": "auto",
              "marginBottom": undefined,
            }
          }
        >
          <a
            href="http://localhost:3000/post/undefined"
          >
            <img
              alt="showcase"
              loading="lazy"
              src="https://res.cloudinary.com/dldiferrn/image/upload/v1667951957/darwin/npkeplnzbl30me7eu80p.png?w=350&fit=crop&auto=format"
              style={
                Object {
                  "height": "26rem",
                  "objectFit": "cover",
                  "width": "26rem",
                }
              }
            />
          </a>
          <div
            className="MuiImageListItemBar-root MuiImageListItemBar-positionTop css-tdboxv-MuiImageListItemBar-root"
          >
            <div
              className="MuiImageListItemBar-titleWrap MuiImageListItemBar-titleWrapTop MuiImageListItemBar-titleWrapActionPosLeft css-3pkysr-MuiImageListItemBar-titleWrap"
            >
              <div
                className="MuiImageListItemBar-title css-dasnyc-MuiImageListItemBar-title"
              >
                   ifdoaspfdsa
              </div>
            </div>
            <div
              className="MuiImageListItemBar-actionIcon MuiImageListItemBar-actionIconActionPosLeft css-1wnvjk9-MuiImageListItemBar-actionIcon"
            >
              <div />
            </div>
          </div>
        </li>
        <li
          className="MuiImageListItem-root MuiImageListItem-standard css-ta95xa-MuiImageListItem-root"
          style={
            Object {
              "gridColumnEnd": "span 1",
              "gridRowEnd": "span 1",
              "height": "auto",
              "marginBottom": undefined,
            }
          }
        >
          <a
            href="http://localhost:3000/post/undefined"
          >
            <img
              alt="showcase"
              loading="lazy"
              src="https://res.cloudinary.com/dldiferrn/image/upload/v1667881383/darwin/yuqitdi6btgswi5grkls.png?w=350&fit=crop&auto=format"
              style={
                Object {
                  "height": "26rem",
                  "objectFit": "cover",
                  "width": "26rem",
                }
              }
            />
          </a>
          <div
            className="MuiImageListItemBar-root MuiImageListItemBar-positionTop css-tdboxv-MuiImageListItemBar-root"
          >
            <div
              className="MuiImageListItemBar-titleWrap MuiImageListItemBar-titleWrapTop MuiImageListItemBar-titleWrapActionPosLeft css-3pkysr-MuiImageListItemBar-titleWrap"
            >
              <div
                className="MuiImageListItemBar-title css-dasnyc-MuiImageListItemBar-title"
              >
                   
              </div>
            </div>
            <div
              className="MuiImageListItemBar-actionIcon MuiImageListItemBar-actionIconActionPosLeft css-1wnvjk9-MuiImageListItemBar-actionIcon"
            >
              <div />
            </div>
          </div>
        </li>
        <li
          className="MuiImageListItem-root MuiImageListItem-standard css-ta95xa-MuiImageListItem-root"
          style={
            Object {
              "gridColumnEnd": "span 1",
              "gridRowEnd": "span 1",
              "height": "auto",
              "marginBottom": undefined,
            }
          }
        >
          <a
            href="http://localhost:3000/post/undefined"
          >
            <img
              alt="showcase"
              loading="lazy"
              src="https://res.cloudinary.com/dldiferrn/image/upload/v1668184748/darwin/ldkbztnskp9kif6vcpyj.png?w=350&fit=crop&auto=format"
              style={
                Object {
                  "height": "26rem",
                  "objectFit": "cover",
                  "width": "26rem",
                }
              }
            />
          </a>
          <div
            className="MuiImageListItemBar-root MuiImageListItemBar-positionTop css-tdboxv-MuiImageListItemBar-root"
          >
            <div
              className="MuiImageListItemBar-titleWrap MuiImageListItemBar-titleWrapTop MuiImageListItemBar-titleWrapActionPosLeft css-3pkysr-MuiImageListItemBar-titleWrap"
            >
              <div
                className="MuiImageListItemBar-title css-dasnyc-MuiImageListItemBar-title"
              >
                   fdafdsafewqreqwrewq
              </div>
            </div>
            <div
              className="MuiImageListItemBar-actionIcon MuiImageListItemBar-actionIconActionPosLeft css-1wnvjk9-MuiImageListItemBar-actionIcon"
            >
              <div />
            </div>
          </div>
        </li>
      </ul>
    </div>
  </section>,
  <div
    className="container"
  >
    <h1>
      ——————————————————————————————
    </h1>
  </div>,
]
`;

exports[`test Profile profile matches snapshot 2`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark bg-primary bg-gradient shadow"
>
  <div
    className="container-fluid"
  >
    <a
      className="navbar-brand"
      href="#"
    >
      <img
        alt=""
        height="40"
        src="logo.png"
        width="115"
      />
    </a>
    <button
      aria-controls="navbarSupportedContent"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-bs-target="#navbarSupportedContent"
      data-bs-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarSupportedContent"
    >
      <ul
        className="navbar-nav me-auto mb-2 mb-lg-0"
      >
        <li
          className="nav-item"
        >
          <a
            aria-current="page"
            className="nav-link active"
            onClick={[Function]}
          >
            Activity
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            aria-current="page"
            className="nav-link active"
            id="search"
            onClick={[Function]}
          >
            Search
          </a>
        </li>
        <li
          className="nav-item dropdown"
        >
          <a
            aria-expanded="false"
            className="nav-link dropdown-toggle"
            data-bs-toggle="dropdown"
            href="#"
            id="navbarDropdown"
            role="button"
          >
            Todo
          </a>
          <ul
            aria-labelledby="navbarDropdown"
            className="dropdown-menu"
          >
            <li>
              <a
                className="dropdown-item"
                onClick={[Function]}
              >
                My Profile
              </a>
            </li>
            <li>
              <a
                className="dropdown-item"
                onClick={[Function]}
              >
                New Post
              </a>
            </li>
            <li>
              <hr
                className="dropdown-divider"
              />
            </li>
            <li>
              <a
                className="dropdown-item"
                onClick={[Function]}
              >
                Log Out
              </a>
            </li>
          </ul>
        </li>
      </ul>
      <span
        className="navbar-text"
      >
        <div
          className="navbar-brand marginleft=100px"
        />
      </span>
      <span
        className="navbar-brand"
      >
        <a
          className="navbar-brand"
          href="http://localhost:3000/profile/null"
        >
          <img
            alt="showcase"
            className="mb-0 rounded-circle border border-light border-2 display: inline"
            style={
              Object {
                "borderRadius": "50%",
                "height": "2.7rem",
                "objectFit": "cover",
                "width": "2.7rem",
              }
            }
          />
        </a>
      </span>
    </div>
  </div>
</nav>
`;

exports[`test Profile profile matches snapshot 3`] = `<mock_api />`;

exports[`test Profile profile matches snapshot 4`] = `
Array [
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-primary bg-gradient shadow"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand"
        href="#"
      >
        <img
          alt=""
          height="40"
          src="logo.png"
          width="115"
        />
      </a>
      <button
        aria-controls="navbarSupportedContent"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-bs-target="#navbarSupportedContent"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        id="navbarSupportedContent"
      >
        <ul
          className="navbar-nav me-auto mb-2 mb-lg-0"
        >
          <li
            className="nav-item"
          >
            <a
              aria-current="page"
              className="nav-link active"
              onClick={[Function]}
            >
              Activity
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              aria-current="page"
              className="nav-link active"
              id="search"
              onClick={[Function]}
            >
              Search
            </a>
          </li>
          <li
            className="nav-item dropdown"
          >
            <a
              aria-expanded="false"
              className="nav-link dropdown-toggle"
              data-bs-toggle="dropdown"
              href="#"
              id="navbarDropdown"
              role="button"
            >
              Todo
            </a>
            <ul
              aria-labelledby="navbarDropdown"
              className="dropdown-menu"
            >
              <li>
                <a
                  className="dropdown-item"
                  onClick={[Function]}
                >
                  My Profile
                </a>
              </li>
              <li>
                <a
                  className="dropdown-item"
                  onClick={[Function]}
                >
                  New Post
                </a>
              </li>
              <li>
                <hr
                  className="dropdown-divider"
                />
              </li>
              <li>
                <a
                  className="dropdown-item"
                  onClick={[Function]}
                >
                  Log Out
                </a>
              </li>
            </ul>
          </li>
        </ul>
        <span
          className="navbar-text"
        >
          <div
            className="navbar-brand marginleft=100px"
          />
        </span>
        <span
          className="navbar-brand"
        >
          <a
            className="navbar-brand"
            href="http://localhost:3000/profile/null"
          >
            <img
              alt="showcase"
              className="mb-0 rounded-circle border border-light border-2 display: inline"
              style={
                Object {
                  "borderRadius": "50%",
                  "height": "2.7rem",
                  "objectFit": "cover",
                  "width": "2.7rem",
                }
              }
            />
          </a>
        </span>
      </div>
    </div>
  </nav>,
  <section
    className="p-5 bg-dark text-light"
  >
    <div
      className="container"
    >
      <div
        className="d-flex"
      >
        <img
          alt="showcase"
          loading="lazy"
          src=""
          style={
            Object {
              "borderRadius": "50%",
              "height": "16rem",
              "objectFit": "cover",
              "width": "16rem",
            }
          }
        />
        <div
          className="mx-4"
        >
          <h1>
            <span
              className="text-warning"
            />
          </h1>
          <div>
            <h4
              className="text-warning"
            >
              follower:
            </h4>
            <h4
              className="text-warning"
            >
              0
            </h4>
          </div>
          <button
            aria-pressed="true"
            autoComplete="off"
            className="btn btn-warning btn-sm active followbtn"
            data-bs-toggle="button"
            id="followbtn"
            onClick={[Function]}
            type="button"
          >
            FOLLOW!
          </button>
          <p
            className="my-4"
          />
        </div>
      </div>
    </div>
    <div
      className="container bg-dark"
    >
      <div />
    </div>
  </section>,
  <div
    className="container"
  >
    <h1
      className="post1"
    >
      Posts:
    </h1>
  </div>,
  <div
    className="container"
  >
     
    <h3>
       :( Dont have any posts !
    </h3>
  </div>,
  <div
    className="container"
  >
    <h1>
      ——————————————————————————————
    </h1>
  </div>,
]
`;
